lib/class-wp-rest-widget-types-controller.php

```diff
@@ -200,19 +201,25 @@ protected function get_widgets() {
 		global $wp_registered_widgets;
 
 		$widgets = array();
-		foreach ( $wp_registered_widgets as  $widget ) {
-			$widget_callback = $widget['callback'];
+
+		foreach ( $wp_registered_widgets as $widget ) {
+			$parsed_id    = gutenberg_parse_widget_id( $widget['id'] );
```

Comment:
What is this ID? Why is it needed?

Reply:
This determines the `id` of the widget type by iterating through `$wp_registered_widgets` and looking at the base name of each widget's ID.

For example, `$wp_registered_widgets` might look like this:

```
array(
	'text-2', => ...
	'text-3', => ...
	'search-1', => ...
	'rss-1', => ...
	'marquee_greeting', => ...
)
```

This code will, from that, calculate that there are four widget types with the following IDs:

```
text
search
rss
marquee_greeting
```

Reply:
Yeah. I don't want to move away from calling it `id`, `id_base` and `number` as it's the phraseology that Core uses pretty extensively. In fact, this whole `parse_widget_id` function is the exact same as `WP_Customize_Widgets::parse_widget_id`. When merged to Core we can de-duplicate.

https://github.com/WordPress/wordpress-develop/blob/master/src/wp-includes/class-wp-customize-widgets.php#L600